tests/foreman/cli/test_bootstrap_script.py

```diff
@@ -0,0 +1,148 @@
+# -*- encoding: utf-8 -*-
+"""Test for bootstrap script"""
+from robottelo.decorators import stubbed
+from robottelo.test import CLITestCase
+
+
+class BootstrapScriptTestCase(CLITestCase):
+    """Test class for bootstrap scipt."""
+
+    @stubbed()
+    @classmethod
+    def setUpClass(cls):
+        """ create VM for testing """
+
+    @stubbed()
+    def test_positive_register(self):
+        """@test: system is registered
+
+        @Feature: Bootstrap Script
+
+        @Setup: none
+
+        @Steps:
+
+        1. assure system is not registered
+        2. register a system
+
+        @Assert: system is registered, host is created
+
+        @Status: Manual
+
+        """
+
+    @stubbed()
+    def test_positive_reregister(self):
+        """@test: registered system is re-registered
+
+        @Feature: Bootstrap Script
+
+        @Setup: none
+
+        @Steps:
+
+        1. register a system
+        2. assure system is registered
+        3. register system once again
+
+        @Assert: system is newly registered, host is created
+
+        @Status: Manual
+
+        """
+
+    @stubbed()
+    def test_positive_migrate(self):
+        """@test: rhn registered system is migrated
+
+        @Feature: Bootstrap Script
+
+        @Setup: none
+
+        @Steps:
+
+        1. register system to SAT5 (or use precreated stored registration)
+        2. assure system is registered with rhn classic
+        3. migrate system
+
+        @Assert: system is migrated, ie. registered
+
+        @Status: Manual
+
+        """
+
+    @stubbed()
+    def test_negative_register_no_subs(self):
+        """@test: tries to register when no subscriptions are available
+
+        @Feature: Bootstrap Script
+
+        @Setup: none
+
+        @Steps:
+
+        1. create env with no available subscriptions
+        2. try to register a system
+
+        @Assert: ends gracefully, reason displayed to user
+
+        @Status: Manual
+
+        """
+
+    @stubbed()
+    def test_negative_register_bad_hostgroup(self):
+        """@test: tries to register when hostgroup doesn't meet all criteria
+
+        @Feature: Bootstrap Script
+
+        @Setup: none
+
+        @Steps:
+
+        1. create hostgroup not matching required criteria for bootstrapping
+        2. try to register a system
+
+        @Assert: ends gracefully, reason displayed to user
+
+        @Status: Manual
+
+        """
+
+    @stubbed()
+    def test_positive_register_host_collision(self):
+        """@test: tries to register with already created host and succeeds?
+
+        @Feature: Bootstrap Script
+
+        @Setup: none
+
+        @Steps:
+
+        1. create host profile
```

Comment:
bootstrap creates host itself, so this is about creating host profile before bootstrap run for possible collision.